pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -531,14 +532,15 @@ public CompletableFuture<Void> messageAcked(CommandAck ack) {
 
     //this method is for individual ack not carry the transaction
     private CompletableFuture<Long> individualAckNormal(CommandAck ack, Map<String, Long> properties) {
-        List<Position> positionsAcked = new ArrayList<>();
+        List<Pair<Consumer, Position>> positionsAcked = new ArrayList<>();
         long totalAckCount = 0;
         for (int i = 0; i < ack.getMessageIdsCount(); i++) {
             MessageIdData msgId = ack.getMessageIdAt(i);
             Position position;
-            long ackedCount = 0;
-            long batchSize = getBatchSize(msgId);
-            Consumer ackOwnerConsumer = getAckOwnerConsumer(msgId.getLedgerId(), msgId.getEntryId());
+            Pair<Consumer, Long> ackOwnerConsumerPair = getAckOwnerConsumer(msgId.getLedgerId(), msgId.getEntryId());
```

Comment:
It's better to rename the method from `getAckOwnerConsumer` to `getAckOwnerConsumerAndBatchSize`. Similarly, rename `ackOwnerConsumer` to `ackOwnerConsumerAndBatchSize` so that it's possible to understand what the `Pair` is holding.